pkg/logger/logger.go

```diff
@@ -32,21 +33,51 @@ var logLevelOrder = map[LogLevel]int{
 type Logger struct {
 	LogLevel LogLevel
 	File     string
+	charm    *log.Logger
 }
 
-func NewLogger(logLevel LogLevel, file string) (*Logger, error) {
+func InitializeLogger(logLevel LogLevel, file string) (*Logger, error) {
+	charm := GetCharmLogger()
+
+	// Set log level
+	charmLevel := log.InfoLevel
+	switch logLevel {
+	case LogLevelTrace:
+		charmLevel = TraceLevel
+	case LogLevelDebug:
+		charmLevel = log.DebugLevel
+	case LogLevelInfo:
+		charmLevel = log.InfoLevel
+	case LogLevelWarning:
+		charmLevel = log.WarnLevel
+	case LogLevelOff:
+		charmLevel = log.FatalLevel + 1 // Set to a level higher than any defined level.
+	}
+	charm.SetLevel(charmLevel)
+
+	if shouldUseCustomLogFile(file) {
+		logFile, err := openLogFile(file)
+		if err != nil {
+			return nil, err
+		}
+		charm = GetCharmLoggerWithOutput(logFile)
+		charm.SetLevel(charmLevel)
+	}
+
 	return &Logger{
 		LogLevel: logLevel,
 		File:     file,
+		charm:    charm,
 	}, nil
 }
 
-func NewLoggerFromCliConfig(cfg schema.AtmosConfiguration) (*Logger, error) {
+// InitializeLoggerFromCliConfig creates a logger based on Atmos CLI configuration.
+func InitializeLoggerFromCliConfig(cfg *schema.AtmosConfiguration) (*Logger, error) {
```

Comment:
The `NewX` naming convention is the standard way to create and return a new instance of a type.